### DISCUSSION TITLE
Include images from gallery

### DISCUSSION THREAD

**Discussion by FarisHijazi** [BlafKing/sd-civitai-browser-plus#188] [Ideas]
I love this tool so much, I'm able to see models right from within A111 !!
and I love that we can see the poster images by the model, but I would also love to see the community gallery for each model. I love the "send to text2img" and I want to be able to do that with gallerys

Any plans to add this? I can code, tell me a general idea of how this would look like and I might be able to make a pull request.

Example, model: https://civitai.com/models/84040/sdxl-unstable-diffusers-yamermix
model images (currently this is the only thing supported by this extension):
![image](https://github.com/BlafKing/sd-civitai-browser-plus/assets/47270244/b05f0793-2d8c-4768-bb4b-b85313f695b2)

gallery (this is what I want to see in automatic1111)
![image](https://github.com/BlafKing/sd-civitai-browser-plus/assets/47270244/45c67d57-f51b-422b-b8ec-19770097b1f4)

**Comment by BlafKing** [comment]
Nice suggestion!
And thanks for the kind words, glad to hear you like the extension! 😁

I agree that would definitely be a good addition so I'll add that to the to-do list!

The only issue currently is that the public API of CivitAI isn't working all too great, it works fine with model requests for the most part, but  the images endpoint of the API would need to be used to add a gallery, and unfortunately it's currently pretty much completely broken.

It only seems to work if you don't supply any query parameters, and even then it fails to return a `nextCursor`, which would be required to load the next page of images.

If any parameters are provided, the API then simply returns `Internal Server Error`
example URLs:
```
"https://civitai.com/api/v1/images" works, but "nextCursor" is "null"
"https://civitai.com/api/v1/images?limit=10" doesn't work at all
```

The only work around as of right now seems to be cloning an RPC request from the browser, but it's definitely not intended to be used this way and also likely not allowed by CivitAI.

I have sent a message to the founder of CivitAI asking some more info about this and if this would be allowed or not, so I'll wait for his response before I start working on this! :)

If it's not allowed, we'll unfortunately have to wait until the images endpoint is working again.

**Reply by FarisHijazi** [reply]
Aha I see, well one page of images is good enough for noe no?
